### ISSUE TITLE
multiGet crash with Hermes on Android

### ISSUE THREAD

**Issue by pavlos256** [react-native-async-storage/async-storage#560]
## Current behavior

When Hermes is enabled, multiGet behaves like it's hanged for about 10 seconds and then the app crashes. It doesn't happen all the time but the more the keys, the easier it is to reproduce. Possibly related to #445.

## Expected behavior

multiGet should be fast and not crash.

## Repro steps

1. Enable Hermes as per https://reactnative.dev/docs/hermes
2. Wait on multiGet with enough keys. You may have to repeat this a few times before the problem manifests itself.

## Environment

- **Platforms tested**:
  - [x] Android
  - [ ] iOS
  - [ ] macOS
  - [ ] Windows
- **AsyncStorage version**: ^1.14.1
- **Expo version**: 
- **Environment**:
```
System:
    OS: Linux 5.11 Arch Linux
    CPU: x64 AMD Ryzen 5
  Binaries:
    Node: 15.11.0
    Yarn: 1.22.10
    npm: 7.5.2
    Watchman: 4.9.0
  SDKs:
    Android SDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.7042882
  Languages:
    Java: 15.0.2
    Python: 3.9.2
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.4 => 0.63.4 
  npmGlobalPackages:
    *react-native*: Not Found
```
- **Logs/Errors that may be relevant**:
```
03-15 13:12:49.733  9611  9657 W libc    : calloc(134217728, 4) failed: returning null pointer
03-15 13:12:49.734  9611  9657 F libc    : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 9657 (mqt_js), pid 9611 (com.example.test)
03-15 13:12:49.783  9719  9719 I crash_dump32: obtaining output fd from tombstoned, type: kDebuggerdTombstone
03-15 13:12:49.784  1853  1853 I /system/bin/tombstoned: received crash request for pid 9657
03-15 13:12:49.784  9719  9719 I crash_dump32: performing dump of process 9611 (target tid = 9657)
03-15 13:12:49.787  9719  9719 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
03-15 13:12:49.787  9719  9719 F DEBUG   : Build fingerprint: 'google/sdk_gphone_x86/generic_x86:10/QSR1.190920.001/5891938:user/release-keys'
03-15 13:12:49.787  9719  9719 F DEBUG   : Revision: '0'
03-15 13:12:49.787  9719  9719 F DEBUG   : ABI: 'x86'
03-15 13:12:49.787  9719  9719 F DEBUG   : Timestamp: 2021-03-15 13:12:49+0200
03-15 13:12:49.787  9719  9719 F DEBUG   : pid: 9611, tid: 9657, name: mqt_js  >>> com.example.test <<<
03-15 13:12:49.787  9719  9719 F DEBUG   : uid: 10136
03-15 13:12:49.787  9719  9719 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0
03-15 13:12:49.787  9719  9719 F DEBUG   : Cause: null pointer dereference
03-15 13:12:49.787  9719  9719 F DEBUG   :     eax 00000000  ebx be1b0880  ecx bd864df4  edx bd864df4
03-15 13:12:49.787  9719  9719 F DEBUG   :     edi bd864f50  esi bde5a380
03-15 13:12:49.787  9719  9719 F DEBUG   :     ebp bd864f18  esp bd864f00  eip bde5a3ae
03-15 13:12:49.834  9719  9719 F DEBUG   : 
03-15 13:12:49.834  9719  9719 F DEBUG   : backtrace:
03-15 13:12:49.834  9719  9719 F DEBUG   :       #00 pc 000193ae  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (facebook::hermes::detail::hermesFatalErrorHandler(void*, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, bool)+46) (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #01 pc 00289c68  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #02 pc 00289b64  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #03 pc 00254922  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #04 pc 00253cdf  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #05 pc 000c4470  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #06 pc 000c4422  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #07 pc 0004b81f  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #08 pc 0004a5eb  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #09 pc 0004ad9a  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #10 pc 0008c3f5  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #11 pc 0006b77c  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #12 pc 000d69ac  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #13 pc 00134453  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #14 pc 00134211  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #15 pc 00134211  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #16 pc 00134211  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #17 pc 001341da  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #18 pc 001341da  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #19 pc 00133ee7  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #20 pc 000c8b5f  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #21 pc 0004e515  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #22 pc 0005412b  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #23 pc 000504b0  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #24 pc 00058625  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #25 pc 00038dba  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #26 pc 00126b10  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #27 pc 0004e515  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #28 pc 0005412b  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #29 pc 000504b0  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #30 pc 00058625  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #31 pc 00038dba  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #32 pc 00126b10  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #33 pc 0004e515  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #34 pc 0005412b  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #35 pc 000504b0  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #36 pc 00058625  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #37 pc 00038dba  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #38 pc 00036f36  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.834  9719  9719 F DEBUG   :       #39 pc 0012681a  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #40 pc 0004e515  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #41 pc 0005412b  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #42 pc 000504b0  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #43 pc 00058625  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #44 pc 00038dba  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #45 pc 000384a3  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #46 pc 0004e575  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #47 pc 0005412b  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #48 pc 000504b0  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #49 pc 00058625  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #50 pc 00038dba  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #51 pc 000384a3  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #52 pc 00037b26  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #53 pc 0001ed9b  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes.so (facebook::hermes::HermesRuntimeImpl::call(facebook::jsi::Function const&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned int)+731) (BuildId: 71597025b677779614fc9c7b5730cf8d7230ce89)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #54 pc 0001c28b  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes-executor-debug.so (BuildId: 9dce540e51aa69e247c03db8c17722fa203a0b70)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #55 pc 00024d3d  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes-executor-debug.so (_ZNK8facebook3jsi8Function4callIJRKdNS0_5ValueEEEES5_RNS0_7RuntimeEDpOT_+141) (BuildId: 9dce540e51aa69e247c03db8c17722fa203a0b70)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #56 pc 00024b33  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libhermes-executor-debug.so (facebook::react::JSIExecutor::invokeCallback(double, folly::dynamic const&)+211) (BuildId: 9dce540e51aa69e247c03db8c17722fa203a0b70)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #57 pc 0009f0fe  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libreactnativejni.so (BuildId: 930803d0b368271f5f896b0cefe51d5c5c2e92b3)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #58 pc 000a036c  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libreactnativejni.so (BuildId: 930803d0b368271f5f896b0cefe51d5c5c2e92b3)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #59 pc 0005fe15  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libreactnativejni.so (BuildId: 930803d0b368271f5f896b0cefe51d5c5c2e92b3)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #60 pc 0004ee42  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libreactnativejni.so (_ZN8facebook3jni6detail13MethodWrapperIMNS_5react15JNativeRunnableEFvvEXadL_ZNS4_3runEvEES4_vJEE8dispatchENS0_9alias_refIPNS1_8JTypeForINS0_11HybridClassIS4_NS3_8RunnableEE8JavaPartESB_vE11_javaobjectEEE+50) (BuildId: 930803d0b368271f5f896b0cefe51d5c5c2e92b3)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #61 pc 0004eda9  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libreactnativejni.so (_ZN8facebook3jni6detail15FunctionWrapperIPFvNS0_9alias_refIPNS1_8JTypeForINS0_11HybridClassINS_5react15JNativeRunnableENS6_8RunnableEE8JavaPartES8_vE11_javaobjectEEEEXadL_ZNS1_13MethodWrapperIMS7_FvvEXadL_ZNS7_3runEvEES7_vJEE8dispatchESE_EESD_vJEE4callEP7_JNIEnvP8_jobject+73) (BuildId: 930803d0b368271f5f896b0cefe51d5c5c2e92b3)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #62 pc 0004ed53  /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/lib/x86/libreactnativejni.so (_ZN8facebook3jni6detail13MethodWrapperIMNS_5react15JNativeRunnableEFvvEXadL_ZNS4_3runEvEES4_vJEE4callEP7_JNIEnvP8_jobject+35) (BuildId: 930803d0b368271f5f896b0cefe51d5c5c2e92b3)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #63 pc 00144f67  /apex/com.android.runtime/lib/libart.so (art_quick_generic_jni_trampoline+71) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #64 pc 0201e642  /memfd:/jit-cache (deleted) (android.os.Handler.handleCallback+66)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #65 pc 0201fa71  /memfd:/jit-cache (deleted) (android.os.Handler.dispatchMessage+65)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #66 pc 020292f2  /memfd:/jit-cache (deleted) (com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage+50)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #67 pc 0013e7d2  /apex/com.android.runtime/lib/libart.so (art_quick_invoke_stub+338) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #68 pc 00149a69  /apex/com.android.runtime/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+281) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #69 pc 00332502  /apex/com.android.runtime/lib/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+386) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #70 pc 0032c19c  /apex/com.android.runtime/lib/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+988) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #71 pc 0068186d  /apex/com.android.runtime/lib/libart.so (MterpInvokeVirtual+989) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #72 pc 00138821  /apex/com.android.runtime/lib/libart.so (mterp_op_invoke_virtual+33) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #73 pc 00319eda  /system/framework/framework.jar (android.os.Looper.loop+466)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #74 pc 00684f6c  /apex/com.android.runtime/lib/libart.so (MterpInvokeStatic+1260) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #75 pc 001389a1  /apex/com.android.runtime/lib/libart.so (mterp_op_invoke_static+33) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #76 pc 002a3fb2  [anon:dalvik-classes.dex extracted in memory from /data/app/com.example.test-lafXk4ekJrWafv-REL2btQ==/base.apk] (com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run+74)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #77 pc 006837bc  /apex/com.android.runtime/lib/libart.so (MterpInvokeInterface+1980) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #78 pc 00138a21  /apex/com.android.runtime/lib/libart.so (mterp_op_invoke_interface+33) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #79 pc 000ea918  /apex/com.android.runtime/javalib/core-oj.jar (java.lang.Thread.run+8)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #80 pc 002f8e0a  /apex/com.android.runtime/lib/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEbb.llvm.12194892193087984976+298) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #81 pc 002ffcc5  /apex/com.android.runtime/lib/libart.so (art::interpreter::EnterInterpreterFromEntryPoint(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*)+181) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #82 pc 0066fbd9  /apex/com.android.runtime/lib/libart.so (artQuickToInterpreterBridge+1209) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #83 pc 0014503d  /apex/com.android.runtime/lib/libart.so (art_quick_to_interpreter_bridge+77) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #84 pc 0013e7d2  /apex/com.android.runtime/lib/libart.so (art_quick_invoke_stub+338) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #85 pc 00149a69  /apex/com.android.runtime/lib/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+281) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #86 pc 0055a513  /apex/com.android.runtime/lib/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+99) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #87 pc 0055b91a  /apex/com.android.runtime/lib/libart.so (art::InvokeVirtualOrInterfaceWithJValues(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, jvalue const*)+474) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #88 pc 005aaa51  /apex/com.android.runtime/lib/libart.so (art::Thread::CreateCallback(void*)+1585) (BuildId: 895645e5113da057f27d9b2ec11eb3bf)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #89 pc 0011a8e5  /apex/com.android.runtime/lib/bionic/libc.so (__pthread_start(void*)+53) (BuildId: 76290498408016ad14f4b98c3ab6c65c)
03-15 13:12:49.835  9719  9719 F DEBUG   :       #90 pc 000af6a7  /apex/com.android.runtime/lib/bionic/libc.so (__start_thread+71) (BuildId: 76290498408016ad14f4b98c3ab6c65c)
03-15 13:12:49.978  1853  1853 E /system/bin/tombstoned: Tombstone written to: /data/tombstones/tombstone_03
03-15 13:12:49.979  2004  2043 I BootReceiver: Copying /data/tombstones/tombstone_03 to DropBox (SYSTEM_TOMBSTONE)
```

**Comment by tido64** [comment]
Can you provide a minimal project with the crash? Does it crash if you _don't_ turn on Hermes? If it doesn't crash on JSC, then I would suggest you open an issue at https://github.com/facebook/hermes instead. Judging from the stack trace alone, it doesn't look like AsyncStorage is the cause here.

**Comment by pavlos256** [comment]
You're right, it doesn't crash on JSC and I confirmed it's a Hermes issue by removing code until I was left with a minimal test case. 

Turns out it's a known issue related to this one (https://github.com/facebook/hermes/issues/95) and it seems to be fixed on 0.64 (I can't make it crash).

**Comment by marcshilling** [comment]
@pavlos256 I'm getting this on 0.64, only with Hermes enabled. Any other ideas?
